conjure_oxide/src/rule_engine/resolve_rules.rs

```diff
@@ -0,0 +1,142 @@
+use conjure_core::rule::Rule;
+use conjure_rules::get_rule_set_by_name;
+use conjure_rules::rule_set::RuleSet;
+use std::collections::{HashMap, HashSet};
+use std::fmt::Display;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum ResolveRulesError {
+    RuleSetNotFound,
+}
+
+impl Display for ResolveRulesError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ResolveRulesError::RuleSetNotFound => write!(f, "Rule set not found."),
+        }
+    }
+}
+
+/**
+* Helper function to get a rule set by name, or return an error if it doesn't exist.
+* @param `rule_set_name` The name of the rule set to get.
+* @returns The rule set with the given name or RuleSetError::RuleSetNotFound if it doesn't exist.
```

Comment:
(Pedantic)

Rustdoc doesn't understand javadoc syntax, just markdown.

For returns a common thing to do is have a section called Returns (https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md#module-level-vs-type-level-docs)

Maybe for arguments you can do 

```
# Arguments
* `rule_set_name` The name of the ruleset to get
```

but this comment is arguably unnecessary here.

Reply:
See also https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html.

You also should probably use /// on each line of a doc comment, that seems to be convention too